components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/handlers/UsernameValidationConfigurationHandler.java

```diff
@@ -0,0 +1,162 @@
+
+            // Return the JsRegex if the default regex has been updated by the user.
+            if (!usernameRegEx.get(JAVA_REG_EX).isEmpty() && !usernameRegEx.get(JS_REG_EX).isEmpty() &&
+                    !DEFAULT_EMAIL_REGEX_PATTERN.equals(usernameRegEx.get(JAVA_REG_EX))) {
+                rules.add(getRuleConfig("JsRegExValidator", JS_REGEX, usernameRegEx.get(JS_REG_EX)));
+                configuration.setRegEx(rules);
+            } else {
+                rules.add(getRuleConfig(EmailFormatValidator.class.getSimpleName(),
+                        ENABLE_VALIDATOR, Boolean.TRUE.toString()));
+                configuration.setRules(rules);
+            }
+            return configuration;
+        } catch (InputValidationMgtException e) {
+            throw new InputValidationMgtException(ERROR_GETTING_EXISTING_CONFIGURATIONS.getCode(), e.getMessage());
+        }
+    }
+
+    @Override
+    public boolean validateValidationConfiguration(List<RulesConfiguration> configurationList)
+            throws InputValidationMgtClientException {
+
+        List<String> validatorNames = new ArrayList<>();
+        configurationList.forEach(config -> validatorNames.add(config.getValidatorName()));
+        int validConfigurations = 0;
+        for (RulesConfiguration configuration: configurationList) {
+            if (EmailFormatValidator.class.getSimpleName().equals(configuration.getValidatorName())) {
+                if (Boolean.parseBoolean(configuration.getProperties().get(ENABLE_VALIDATOR))) {
+                    validConfigurations += 1;
+                }
+                validatorNames.remove(EmailFormatValidator.class.getSimpleName());
+            } else if (AlphanumericValidator.class.getSimpleName().equals(configuration.getValidatorName())) {
+                if (Boolean.parseBoolean(configuration.getProperties().get(ENABLE_VALIDATOR))) {
+                    if (validatorNames.contains(LengthValidator.class.getSimpleName())) {
+                        validConfigurations += 1;
+                        validatorNames.remove(LengthValidator.class.getSimpleName());
+                    }
+                }
+                validatorNames.remove(AlphanumericValidator.class.getSimpleName());
+            }
+        }
+        if (validConfigurations != 1 || !validatorNames.isEmpty()) {
+            throw new InputValidationMgtClientException(ERROR_INVALID_VALIDATORS_COMBINATION.getCode(),
+                    String.format(ERROR_INVALID_VALIDATORS_COMBINATION.getDescription(), USERNAME));
+        }
+        return true;
+    }
+
+    private Map<String, String> getUsernameRegEx(RealmConfiguration realmConfig) {
+
+        Map<String, String> usernameRegEx = new HashMap<>();
+        if (MultitenantUtils.isEmailUserName()) {
+
+            if (StringUtils.isNotBlank(realmConfig.getUserStoreProperty(PROPERTY_USER_NAME_WITH_EMAIL_JS_REG_EX))) {
+
+                usernameRegEx.put(JAVA_REG_EX, realmConfig.getUserStoreProperty(
```

Comment:
I don't see much requirement of having two values in this hash map, one value with '/' and the other value with '//'. Please validate with he below conditions as well. (134-159)

Reply:
Addressed